libraries/AP_Compass/AP_Compass.h

```diff
@@ -333,10 +336,20 @@ friend class AP_Compass_Backend;
 
 private:
     static Compass *_singleton;
+
+    // Use Priority and StateIndex typesafe index types 
+    // to distinguish between two different type of indexing
+    // We use StateIndex for access by Backend
+    // and Priority for access by Frontend
+    DECLARE_TYPESAFE_INDEX(Priority, int);
+    DECLARE_TYPESAFE_INDEX(StateIndex, int);
```

Comment:
We don't usually use camel case except for enums.  Priority and StateIndex are objects I guess which normally would use the regular lower-case-with-underlines-between-words style.

Reply:
They are not objects but types, closer to a class. Making them lower-case-with-underlines-between-words will make developer confuse them with a function which they are not when using as a constructor. I would like them to stand out in code, as they play a very important roll of not allowing the two types of indexes to be confused with each other or used instead of each other.